lib/datasource.js

```diff
@@ -433,19 +433,19 @@ DataSource.prototype.defineRelations = function(modelClass, relations) {
   var self = this;
 
   // Create a function for the closure in the loop
-  var createListener = function(name, relation, targetModel, throughModel) {
+  var createListener = function(relationName, relation, targetModel, throughModel) {
```

Comment:
- renamed property `name` to `relationName` to make it more readable
- i also propose to modify the function name from `createListener()` to something like `deferRelationSetup()`